cost-analyzer/templates/NOTES.txt

```diff
@@ -1,9 +1,35 @@
 
 
 --------------------------------------------------
+{{- $node := (lookup "v1" "Node" "" "") }}
+{{- $isEKS := (regexMatch ".*eks.*" (.Capabilities.KubeVersion | quote) )}}
+{{- $isGT22 := (semverCompare ">=1.23-0" .Capabilities.KubeVersion.GitVersion) }}
+{{- $PVNotExists := (empty (lookup "v1" "PersistentVolume" "" "")) }}
+{{- $EBSCSINotExists := (empty (lookup "apps/v1" "Deployment" "kube-system" "ebs-csi-controller")) }}
+
 {{- $servicePort := .Values.service.port | default 9090 -}}
 Kubecost has been successfully installed.
 
+{{ if (and $isEKS $isGT22) -}}
+
+WARNING: ON EKS v1.23+ INSTALLATION OF EBS-CSI DRIVER IS REQUIRED TO MANAGE PERSISTENT VOLUMES. LEARN MORE HERE: https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html
+
+{{ if (and $EBSCSINotExists $PVNotExists) -}}
+
+ERROR: MISSING EBS-CSI DRIVER WHICH IS REQUIRED ON EKS v1.23+ TO MANAGE PERSISTENT VOLUMES. LEARN MORE HERE: https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html
+
+{{ else if (and $EBSCSINotExists (not $PVNotExists)) -}}
+
+ERROR: MISSING EBS-CSI DRIVER WHICH IS REQUIRED ON EKS v1.23+ TO MANAGE PERSISTENT VOLUMES. LEARN MORE HERE: https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html
```

Comment:
Should we stop install process on such a clear error? @linhlam-kc @kwombach12

Reply:
We can do that with a "fail" function, just leaving as a note once we decide:

https://stackoverflow.com/questions/61496709/helm-template-how-raise-exception-in-in-helm-function

Reply:
IMO, we should not put a hard stop here but add more details of the recommended action. E.g:
`TO MANAGE PERSISTENT VOLUMES. LEARN MORE HERE: https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html OR TRY OUR FOLLOWING DOCUMENTATION HERE: https://docs.kubecost.com/install-and-configure/install/provider-installations/aws-eks-cost-monitoring#prerequisites` . If there is feedback from users that they absolutely wants to have a hard stop here, we can implement it later.

Reply:
My only concern is that users miss this Error message and are still confused... maybe we make `ERROR` bold & red?

Reply:
Afaik, the NOTES is plain text, so I'm not sure if we can format the texts. 

On the note of whether we should put a hard stop. My concern is that putting a hard stop may block users from using Kubecost, if they have EBS driver installed in non-standard way or some use cases that we are not covered by our logic. It looks like an one-way-door to me that may require us to release a hot patch or users has to wait until the next release of our chart in that case.

In addition, I want users to see this as an Amazon EKS limitation but not our product limitation.